### PR TITLE
[Fix #309] `am` uses alias for fully qualified ns

### LINKED ISSUE #309: Add-missing-libspec creates alias for fully qualified namespaces
When doing <kbd> am </kbd> on `schema.test/validate-schemas` I got `[schema.test :as schema.test]`.  While not technically wrong, it's slightly retarded.

### PR DESCRIPTION
**Pull request by expez**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 ### Bugs fixed
 
+- [#309](https://github.com/clojure-emacs/clj-refactor.el/issues/309) `*data-readers*` `am` creates alias for fully-qualified symbols.
 - [#320](https://github.com/clojure-emacs/clj-refactor.el/issues/320) `*data-readers*` ignored when searching for macros.
```
Comment (benedekfazekas) [inline CHANGELOG.md]: copy/paste typo `*data-readers*`

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 ### Bugs fixed
 
+- [#309](https://github.com/clojure-emacs/clj-refactor.el/issues/309) `*data-readers*` `am` creates alias for fully-qualified symbols.
```
Comment (benedekfazekas) [inline CHANGELOG.md]: copy/paste typo `*data-readers*`

File: clj-refactor.el [hunk 3]
```diff
@@ -2970,15 +2970,21 @@ Inspect SYMBOL, the thing at point, to find out whether we have
...
-      (if alias?
-          (cljr--insert-libspec-verbosely (format "[%s :as %s]" missing
-                                                  (cljr--symbol-prefix symbol)))
-        (if (and (s-contains-p "." missing)
-                 (s-uppercase? (s-left 1 (cljr--symbol-suffix missing))))
-            (cljr--insert-libspec-verbosely  (cljr--symbol-prefix symbol))
-          (cljr--insert-libspec-verbosely (format "[%s :refer [%s]]"
+    (let* ((missing (format "%s" missing-symbol))
+           (alias? (cljr--qualified-symbol-p symbol)))
+      (cond
```
Comment (benedekfazekas) [inline clj-refactor.el]: why the `let*`?

File: clj-refactor.el [hunk 4]
```diff
@@ -2970,15 +2970,21 @@ Inspect SYMBOL, the thing at point, to find out whether we have
...
-          (alias? (cljr--qualified-symbol-p symbol)))
-      (if alias?
-          (cljr--insert-libspec-verbosely (format "[%s :as %s]" missing
-                                                  (cljr--symbol-prefix symbol)))
-        (if (and (s-contains-p "." missing)
-                 (s-uppercase? (s-left 1 (cljr--symbol-suffix missing))))
-            (cljr--insert-libspec-verbosely  (cljr--symbol-prefix symbol))
-          (cljr--insert-libspec-verbosely (format "[%s :refer [%s]]"
+    (let* ((missing (format "%s" missing-symbol))
+           (alias? (cljr--qualified-symbol-p symbol)))
```
Comment (benedekfazekas) [inline clj-refactor.el]: why the `let*`?

File: clj-refactor.el [hunk 5]
```diff
@@ -2993,8 +2999,8 @@ to create an alias or refer."
            ;; In the line below we're assuming that all clojure code
            ;; will prefer - over _ when naming namespaces :(
            (progn (cljr--insert-missing-require
-                   (s-replace "_" "-" (format "%s" missing-symbol))
-                   missing-symbol)
+                   symbol
+                   (s-replace "_" "-" (format "%s" missing-symbol)))
                   (cljr--insert-missing-import missing-symbol)))
           ((eq type :class) (cljr--insert-missing-import missing-symbol))
           (t (error (format "Uknown type %s" type))))))
```
Comment (benedekfazekas) [inline clj-refactor.el]: typo: `Uknown`

File: CHANGELOG.md (hunk 2 above)
Comment (expez) [inline CHANGELOG.md]: Indeed

File: clj-refactor.el (hunk 4 above)
Comment (expez) [inline clj-refactor.el]: Leftover from earlier draft

**Pull request merged**